lib/tapyrus/tip0137.rb

```diff
@@ -0,0 +1,131 @@
+      address: nil,
+      data: nil,
+      client: nil
+    )
+      validate_payload!(
+        txid: txid,
+        index: index,
+        color_id: color_id,
+        value: value,
+        script_pubkey: script_pubkey,
+        address: address,
+        data: data
+      )
+      if client
+        validate_on_blockchain!(
+          client: client,
+          txid: txid,
+          index: index,
+          color_id: color_id,
+          value: value,
+          script_pubkey: script_pubkey
+        )
+      end
+      data = {
+        txid: txid,
+        index: index,
+        color_id: color_id&.to_hex,
+        value: value,
+        script_pubkey: script_pubkey&.to_hex,
+        address: address,
+        data: data
+      }
+      JWS.encode(data, key.priv_key)
+    end
+
+    #   # @param jws [String] JWT Web Token
+    #   # @param key [Tapyrus::Key] public key
+    #   # @param client [Tapyrus::RPC::TapyrusCoreClient] rpc client
+    #   # @return decoded JSON Object
+    def verify_message(jws, key, client: nil)
+      JWS
+        .decode(jws, key.pubkey)
+        .tap do |decoded|
+          payload = decoded[0]
+          color_id =
+            begin
+              Tapyrus::Color::ColorIdentifier.parse_from_payload(payload['color_id']&.htb)
+            rescue => _e
+              raise RuntimeError, 'color_id is invalid'
+            end
+          script_pubkey =
+            begin
+              Tapyrus::Script.parse_from_payload(payload['script_pubkey']&.htb)
+            rescue => _e
+              raise RuntimeError, 'script_pubkey is invalid'
+            end
+          validate_payload!(
+            txid: payload['txid'],
+            index: payload['index'],
+            color_id: color_id,
+            value: payload['value'],
+            script_pubkey: script_pubkey,
+            address: payload['address'],
+            data: payload['data']
+          )
+          if client
+            validate_on_blockchain!(
+              client: client,
+              txid: payload['txid'],
+              index: payload['index'],
+              color_id: color_id,
+              value: payload['value'],
+              script_pubkey: script_pubkey
+            )
+          end
+        end
+    end
+
+    def validate_payload!(txid: nil, index: nil, color_id: nil, value: nil, script_pubkey: nil, address: nil, data: nil)
+      raise RuntimeError, 'txid is invalid' if !txid || !/^[0-9a-fA-F]{64}$/.match(txid)
+      raise RuntimeError, 'index is invalid' if !index || !/^\d+$/.match(index.to_s) || index < 0 || index >= 2**32
+      if !color_id || !color_id.is_a?(Tapyrus::Color::ColorIdentifier) || !color_id.valid?
+        raise RuntimeError, 'color_id is invalid'
+      end
+      raise RuntimeError, 'value is invalid' if !value || !/^\d+$/.match(value.to_s) || value < 0 || value >= 2**64
+      if !script_pubkey || !script_pubkey.is_a?(Tapyrus::Script) || !/^([0-9a-fA-F]{2})+$/.match(script_pubkey.to_hex)
```

Comment:
`!/^([0-9a-fA-F]{2})+$/.match(script_pubkey.to_hex)`

Why this check necessary?

Reply:
It is not necessary. remove this code (https://github.com/chaintope/tapyrusrb/commit/db2504df92c347691a3850571663f1f708793bad)